enhancements/authentication/pod-security-admission-autolabeling.md

```diff
@@ -0,0 +1,290 @@
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - TBD
+see-also:
+  - "/enhancements/authentication/pod-security-admission.md"
+replaces: []
+superseded-by: []
+---
+
+# Pod Security Admission Autolabeling
+
+## Summary
+
+This enhancement expands the "PodSecurity admission in OpenShift". It introduces
+an opt-in mechanism that allows users to to keep their workloads running when
+Pod Security Admission plugin gets turned on.
+
+## Motivation
+
+We want to adhere with the upstream pod security standards for our workloads but
+we also want to provide our users access to the Security Context Constraints
+(hereinafter SCCs) API that they are already used to. However, each of these
+admission plugins works a bit differently and so there must be a middle-man
+that synchronizes the privileges SCCs provide into privileges that Pod Security
+admission (hereinafter PSa) understands.
+
+### Goals
+
+1. OpenShift clusters can run with the restricted Pod Security admission level by default
+without breaking pod controller workloads
+2. Privileged users can opt-in/opt-out a namespace to and from the Pod Security admission
+
+### Non-Goals
+
+1. Make sure that bare pods (pods w/o pod controllers) continue working
+
+## Proposal
+
+### User Stories
+
+1. I would like my workloads to be admitted based on the current pod security standards
+
+### Pod Security Admission Label Synchronization
+
+This section explains how pod security admission works and later elaborates on how to use
+the pod security labeling in order to synchronize SCC permissions into pod security admission
+restriction levels.
+
+The SCC permissions to PSa levels transformation gets explained at the end of this section.
+
+#### Pod Security Admission Introduction
+
+Pod Security admission validates pods' security and distinguishes three different levels:
+- privileged - most privileged mode, anything is allowed
+- baseline   -  minimally restrictive policy which prevents known privilege escalations
+- restricted - heavily restricted policy, following current Pod hardening best practices
+
+By default, there is a cluster-global configuration which enforces the configured policies
+on pods and known workloads.
+
+It is possible to override the cluster-global policy enforce configuration on a per-namespace
+basis by using the `pod-security.kubernetes.io/enforce` label on given namespaces. It is also
+possible to exampt certain users, namespace and runtime classes from the admission completely.
+
+#### PSa Label Synchronization Controller
+
+This section describes a controller that is going to be a part of the cluster-policy-controller.
+
+In order to keep SCCs working alongside restricted cluster-wide PSa level, there must be
+a mechanism that would synchronize the PSa policy level for a namespace that workload runs
+in so that the workload is allowed by both SCC and PSa admissions. For that purpose, a PSa
```

Comment:
```suggestion
in, so that the workload is allowed by both SCC and PSa admissions. For that purpose, a PSa
```

Reply:
:+1: this finally made me look into English grammar for comma usage, until now I felt like commas were used somewhat random
https://webapps.towson.edu/ows/modulecomma.htm